cider/exec/nextgen/Translator.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef CIDER_EXEC_NEXTGEN_TRANSLATOR_H
+#define CIDER_EXEC_NEXTGEN_TRANSLATOR_H
+
+#include "exec/nextgen/OpNode.h"
+
+namespace cider::exec::nextgen {
+
+class Context;
+class JITTuple;
+
+/// \brief A translator will generate code for an OpNode
+///
+/// A translator is associated to a Context and consume a JITTuple, the generate code will
+/// save to the Context
+class Translator {
+ public:
+  Translator() = default;
+  virtual ~Translator() = default;
+
+  virtual void consume(const JITTuple& input, Context* context) {}
+
+ protected:
+  std::shared_ptr<OpNode> opNode_;
+  std::shared_ptr<Translator> successor_;
```

Comment:
`toTranslator` returns `std::unique_ptr<Translator>`, should we align them? 
And need a `setSuccessor` method or constrcutor will construct `successor_`